cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -137,45 +151,116 @@ public void configure(Map<String, ?> configs) {
 
-    return new NewTopic(cruiseControlMetricsTopic, cruiseControlMetricsTopicNumPartition, cruiseControlMetricsTopicReplicaFactor);
+    Map<String, String> config = new HashMap<>(2);
+    config.put(LogConfig.RetentionMsProp(),
+               Long.toString(reporterConfig.getLong(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_RETENTION_MS_CONFIG)));
+    config.put(LogConfig.CleanupPolicyProp(), CRUISE_CONTROL_METRICS_TOPIC_CLEAN_UP_POLICY);
+    newTopic.configs(config);
+    return newTopic;
   }
 
-  private void createCruiseControlMetricsTopic() {
+  protected void createCruiseControlMetricsTopic() {
     try {
       final CreateTopicsResult createTopicsResult = _adminClient.createTopics(Collections.singletonList(_newTopic));
       createTopicsResult.values().get(_newTopic.name()).get();
-      LOG.info("Cruise Control metrics topic created: {}", _cruiseControlMetricsTopic);
+      LOG.info("Cruise Control metrics topic {} is created.", _cruiseControlMetricsTopic);
+    } catch (InterruptedException | ExecutionException e) {
+      LOG.warn("Unable to create Cruise Control metrics topic {}.", _cruiseControlMetricsTopic, e);
+    }
+  }
+
+  protected void maybeUpdateCruiseControlMetricsTopic() {
+    maybeUpdateTopicConfig();
+    maybeIncreaseTopicPartitionCount();
+  }
+
+  protected void maybeUpdateTopicConfig() {
+    try {
+      // Retrieve topic config to check and update.
+      ConfigResource topicResource = new ConfigResource(ConfigResource.Type.TOPIC, _cruiseControlMetricsTopic);
+      DescribeConfigsResult describeConfigsResult = _adminClient.describeConfigs(Collections.singleton(topicResource));
+      Config topicConfig = describeConfigsResult.values().get(topicResource).get();
+      Set<AlterConfigOp> configsToBeSet = new HashSet<>(2);
+      maybeUpdateConfig(configsToBeSet,
+                        LogConfig.RetentionMsProp(),
+                        _newTopic.configs().get(LogConfig.RetentionMsProp()),
+                        topicConfig);
+      maybeUpdateConfig(configsToBeSet,
+                        LogConfig.CleanupPolicyProp(),
+                        _newTopic.configs().get(LogConfig.CleanupPolicyProp()),
+                        topicConfig);
+      if (!configsToBeSet.isEmpty()) {
+        AlterConfigsResult alterConfigsResult = _adminClient.incrementalAlterConfigs(Collections.singletonMap(topicResource, configsToBeSet));
+        alterConfigsResult.values().get(topicResource).get();
+      }
     } catch (InterruptedException | ExecutionException e) {
-      if (!(e.getCause() instanceof TopicExistsException)) {
-        LOG.error("Unable to create Cruise Control topic", e);
+      LOG.warn("Unable to update config of Cruise Cruise Control metrics topic {}", _cruiseControlMetricsTopic, e);
+    }
+  }
+
+  protected void maybeUpdateConfig(Set<AlterConfigOp> configsToAlter,
```

Comment:
Can we (1) make it `static`? ... -> `protected static void maybeUpdateConfig` and (2) move it to `CruiseControlMetricsUtils` with a JavaDoc?